DynamicTablesPkg/Library/FdtHwInfoParserLib/RiscV/HartInfo/RiscVHartInfoDispatcher.c

```diff
@@ -0,0 +1,415 @@
+    return Status;
+  }
+
+  ZeroMem (&CmoInfo, sizeof (CM_RISCV_CMO_NODE));
+  Prop = fdt_getprop (Fdt, CpuNode, "riscv,cbom-block-size", &Len);
+  if (Prop) {
+    CmoInfo.CbomBlockSize = CmoGetBlockSize (fdt32_to_cpu (ReadUnaligned32 ((const UINT32 *)Prop)));
+  } else {
+    DEBUG (
+      (
+       DEBUG_VERBOSE,
+       "%a: Failed to parse cpu node: riscv,cbom-block-size\n",
+       __func__
+      )
+      );
+    return EFI_NOT_FOUND;
+  }
+
+  Prop = fdt_getprop (Fdt, CpuNode, "riscv,cboz-block-size", &Len);
+  if (Prop) {
+    CmoInfo.CbozBlockSize = CmoGetBlockSize (fdt32_to_cpu (ReadUnaligned32 ((const UINT32 *)Prop)));
+  } else {
+    DEBUG (
+      (
+       DEBUG_VERBOSE,
+       "%a: Failed to parse cpu node: riscv,cboz-block-size\n",
+       __func__
+      )
+      );
+  }
+
+  Prop = fdt_getprop (Fdt, CpuNode, "riscv,cbop-block-size", &Len);
+  if (Prop) {
+    CmoInfo.CbopBlockSize = CmoGetBlockSize (fdt32_to_cpu (ReadUnaligned32 ((const UINT32 *)Prop)));
+  } else {
+    DEBUG (
+      (
+       DEBUG_VERBOSE,
+       "%a: Failed to parse cpu node: riscv,cbop-block-size\n",
+       __func__
+      )
+      );
+  }
+
+  // Add the CmObj to the Configuration Manager.
+  Status = AddSingleCmObj (
+             FdtParserHandle,
+             CREATE_CM_RISCV_OBJECT_ID (ERiscVObjCmoInfo),
+             &CmoInfo,
+             sizeof (CM_RISCV_CMO_NODE),
+             NULL
+             );
+  if (EFI_ERROR (Status)) {
+    ASSERT (0);
+  }
+
+  return Status;
+}
+
+/** Create ISA string Info structure
+
+  @param [in]  FdtParserHandle    A handle to the parser instance.
+  @param [in]  CpusNode           cpus node.
+**/
+STATIC
+EFI_STATUS
+EFIAPI
+RiscVIsaStringInfoParser (
+  IN  CONST FDT_HW_INFO_PARSER_HANDLE  FdtParserHandle,
+  IN        INT32                      CpusNode
+  )
+{
+  CM_RISCV_ISA_STRING_NODE  IsaStringInfo;
+  CONST UINT32              *Prop;
+  EFI_STATUS                Status;
+  INT32                     Len, CpuNode;
+  INT32                     AlignedLen;
+  VOID                      *Fdt;
+
+  Fdt     = FdtParserHandle->Fdt;
+  CpuNode = CpusNode;
+  Status  = FdtGetNextNamedNodeInBranch (Fdt, CpusNode, "cpu", &CpuNode);
+  if (EFI_ERROR (Status)) {
+    ASSERT (0);
+    if (Status == EFI_NOT_FOUND) {
+      // Should have found the node.
+      Status = EFI_ABORTED;
+    }
+
+    return Status;
+  }
+
+  // Parse the "cpu" node.
+  if (!FdtNodeIsCompatible (Fdt, CpuNode, &CpuCompatibleInfo)) {
+    ASSERT (0);
+    Status = EFI_ABORTED;
+    return Status;
+  }
+
+  ZeroMem (&IsaStringInfo, sizeof (CM_RISCV_ISA_STRING_NODE));
+  Prop = fdt_getprop (Fdt, CpuNode, "riscv,isa", &Len);
+  if (!Prop) {
+    DEBUG (
+      (
+       DEBUG_ERROR,
+       "%a: Failed to parse cpu node: riscv,isa\n",
+       __func__
+      )
+      );
+    ASSERT (0);
+    return EFI_ABORTED;
+  }
+
+  AlignedLen              = ALIGN_VALUE (Len, 2);
+  IsaStringInfo.IsaString = AllocateZeroPool (AlignedLen);

```

Comment:
Would it be possible to use a (reasonably) large array for the name instead ?
I.e. have this:
```
CHAR8 IsaString[XXX];
instead of:
CHAR8 *IsaString;
```
in `CM_RISCV_ISA_STRING_NODE` ?